src/ModelEvaluationOfflineStiffness.cpp

```diff
@@ -0,0 +1,282 @@
+        do{
+            auto frame(ModelEvaluationBase<Logger>::getLmtFromInputQueue());
+            if (!frame.data.empty())
+                lmtDataFromQueue_.push_back(frame);
+            else runCondition = false;
+        } while (runCondition);
+
+
+        runCondition = ModelEvaluationBase<Logger>::externalTorquesAvailable();
+        //read external torques
+        do{
+            auto frame(ModelEvaluationBase<Logger>::getExternalTorquesFromInputQueue());
+            if (!frame.data.empty())
+                externalTorquesDataFromQueue_.push_back(frame);
+            else runCondition = false;
+        } while (runCondition);
+
+
+        runCondition = true;
+        //read moment arms data
+        do{
+            vector< ceinms::InputConnectors::FrameType > maFrame;
+            for (unsigned i = 0; i < noDof_; ++i) {
+                auto currentMomentArms(ModelEvaluationBase<Logger>::getMomentArmsFromInputQueue(i));
+                if (!currentMomentArms.data.empty())
+                    maFrame.push_back(currentMomentArms);
+                else runCondition = false;
+            }
+            if (runCondition)
+                maDataFromQueue_.push_back(maFrame);
+        } while (runCondition);
+
+        runCondition=ModelEvaluationBase<Logger>::momentArmDerivativesAvailable();
+        //read moment arm derivatives data
+        do{
+            vector< ceinms::InputConnectors::FrameType > dMaFrame;
+            for (unsigned i = 0; i < noDof_; ++i) {
+                auto currentMomentArmDerivs(ModelEvaluationBase<Logger>::getMomentArmDerivativesFromInputQueue(i));
+                if (!currentMomentArmDerivs.data.empty())
+                    dMaFrame.push_back(currentMomentArmDerivs);
+                else runCondition = false;
+            }
+            if (runCondition)
+                dMaDataFromQueue_.push_back(dMaFrame);
+        } while (runCondition);
+    }
+
+    template <typename NMSmodelT, typename Logger>
+    void ModelEvaluationOfflineStiffness<NMSmodelT, Logger>::initOfflineCurve() {
+
+        auto lmtIt(lmtDataFromQueue_.begin());
+        for (auto emgIt : emgDataFromQueue_) {
+            double emgTime = emgIt.time + globalEmDelay_;
+            subject_.setTime(emgTime);
+            subject_.setEmgs(emgIt.data);
+            if (lmtIt == lmtDataFromQueue_.begin()) subject_.updateActivations();
+            if (lmtIt != lmtDataFromQueue_.end() && TimeCompare::lessEqual(lmtIt->time, emgTime)) {
+                subject_.updateActivations();
+                subject_.setTime(lmtIt->time);
+                subject_.setMuscleTendonLengths(lmtIt->data);
+                subject_.updateFibreLengths_OFFLINEPREP();
+                ++lmtIt;
+            }
+            subject_.pushState();
+        }
+        subject_.updateFibreLengthTraces();
+    }
+
+
+    template <typename NMSmodelT, typename Logger>
+    void ModelEvaluationOfflineStiffness<NMSmodelT, Logger>::operator()() {
```

Comment:
Instead of having `ModelEvaluationOfflineStiffness` can't we add the stiffness to the other evaluators? If the derivatives of the moment arms are provided as input it calculates the stiffness, otherwise it doesn't.. What do you think?